src/Microsoft.AspNet.OData/Formatter/Deserialization/ODataDeserializerContext.cs

```diff
@@ -25,8 +26,14 @@ public HttpRequestMessage Request
             set
             {
                 _request = value;
-                InternalRequest = _request != null ? new WebApiRequestMessage(_request) : null;
+                WebApiRequestMessage webApiRequestMessage = _request != null ? new WebApiRequestMessage(_request) : null;
+                InternalRequest = webApiRequestMessage;
                 InternalUrlHelper = _request != null ? new WebApiUrlHelper(_request.GetUrlHelper()) : null;
+
+                // We add this setting via CompatibilityOptions
+                // routeBuilder.SetCompatibilityOptions(CompatibilityOptions.DisableCaseInsensitiveRequestPropertyBinding);
+                CompatibilityOptions options = webApiRequestMessage.Configuration.GetCompatibilityOptions();
```

Comment:
>webApiRequestMessage.Configuration.GetCompatibilityOptions(); [](start = 47, length = 61)

This line will still throw a null ref exception if the user tries to set Request to null.